barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_recursive_flavor.hpp

```diff
@@ -317,6 +317,43 @@ template <typename BuilderType> class UltraRecursiveFlavor_ {
             this->lagrange_first = Commitment::from_witness(builder, native_key->lagrange_first);
             this->lagrange_last = Commitment::from_witness(builder, native_key->lagrange_last);
         };
+
```

Comment:
new deserialization of recursive vkey from field elements. Note that the witness values for circuit_size, num_public_inputs, and pub_inputs_offset are just discarded and their native values are stored instead. This probably deserves further scrutiny, but I don't think they have to be witness values because the proof contains them as witness values anyway. We should figure this out more clearly at some point.